src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -175,17 +167,14 @@ public String getFilename() {
 
         map.add("file", contentsAsResource);
 
-        HttpEntity<MultiValueMap<String, Object>> requestEntity = new HttpEntity<>(map, requestHeaders);
-
-        try {
-            ResponseEntity<UUID> responseEntity = caseServerRest.exchange("/" + CASE_API_VERSION + "/cases/private",
-                    HttpMethod.POST,
-                    requestEntity,
-                    UUID.class);
-            return responseEntity.getBody();
-        } catch (HttpStatusCodeException e) {
-            throw new StudyException("importCase " + e.getStatusCode() + " : " + e.getResponseBodyAsString(), e);
-        }
+        String uuid = webClient.post()
+                .uri(caseServerBaseUri + "/" + CASE_API_VERSION + "/cases/private")
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.MULTIPART_FORM_DATA.toString())
+                .body(BodyInserters.fromValue(map))
+                .retrieve()
+                .bodyToMono(String.class)
+                .block();
```

Comment:
This is forbidden!

Reply:
I'm migarting to webFlux smoothly, so the first step is to migrate all restTemplate http requests to webClient which is non blocking client 
the .block is temporary and will be obviously removed in the next steps.